src/Uno.UWP/Storage/Pickers/FileOpenPicker.cs

```diff
@@ -98,6 +99,37 @@ private void ValidateConfiguration()
 				throw new InvalidOperationException("You must provide at least a general file type filter ('*')");
 			}
 		}
+
+		internal void SetMultipleFileLimit(int limit)
+		{
+			_multipleFileLimit = limit;
+		}
+#endif
+	}
+
+	public static class FileOpenPickerExtensions
+	{
+#if __IOS__
+		/// <summary>
+		/// Sets the file limit a user can select when picking multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+		public static void SetLimit(this FileOpenPicker picker, int limit)
+		{
+			picker.SetMultipleFileLimit(limit);
+		}
+
+		/// <summary>
+		/// Shows the file picker so that the user can pick multiple files.
+		/// </summary>
+		/// <param name="limit">The maximum number of files that the user can pick.</param>
+		/// <returns>When the call to this method completes successfully, it returns a <see cref="FilePickerSelectedFilesArray"/> object that contains
+		/// all the files that were picked by the user. Picked files in this array are represented by <see cref="StorageFile"/> objects.</returns>
+		public static IAsyncOperation<IReadOnlyList<StorageFile>> PickMultipleFilesAsync(this FileOpenPicker picker, int limit)
```

Comment:
I don't think we need this one, most apps will continue to use the original method, with only a call to `SetMultipleFilesLimit` before that.

I'm still wondering if we should make `SetMultipleFilesLimit` cross targeted already, making it a noop for platforms that don't support it, but with a "notimplemented" attribute on it so developers know it won't work when they use it.

Reply:
Removed the method. I will look into making it visible for all platforms but adding the NotImplemented for the rest.